### PR TITLE
Initial work on evolution of ice thickness.

### PR DESCRIPTION
**Pull request by mperego**
This PR:
Modifies StokesFOThickness problem to enable thickness evolution using Tempus implicit schemes.

Adds a few fixes for Albany to run when `MESH_DEPENDS_ON_SOLUTION` is on. 
Adds two tests computing thickness evolution and sensitivities (sensitivity test is enabled when `MESH_DEPENDS_ON_SOLUTION` is on).

### CODE REVIEW

File: src/landIce/evaluators/LandIce_UpdateZCoordinate_Def.hpp [hunk 1]
```diff
@@ -22,15 +22,19 @@
...
+template<typename EvalT, typename Traits, typename ScalarT>
+UpdateZCoordinateMovingTopBase<EvalT, Traits, ScalarT>::
+UpdateZCoordinateMovingTopBase (const Teuchos::ParameterList& p,
                             const Teuchos::RCP<Albany::Layouts>& dl) :
   coordVecIn (p.get<std::string> ("Old Coords Name"), dl->vertices_vector),
   bedTopo(p.get<std::string> ("Bed Topography Name"), dl->node_scalar),
   topSurface(p.get<std::string>("Top Surface Name"), dl->node_scalar),
   coordVecOut(p.get<std::string> ("New Coords Name"), dl->vertices_vector)
 {
+  bool lossyConversion = p.isParameter("Allow Loss Of Derivative Terms") ? p.get<bool>("Allow Loss Of Derivative Terms") : false;
```
Comment (bartgol) [inline src/landIce/evaluators/LandIce_UpdateZCoordinate_Def.hpp:34]: What is this about? Are we planning to support lossy and non-lossy later on? Also, what's the loss we're talking about?

**Review by bartgol — APPROVED** [review]

File: src/landIce/evaluators/LandIce_UpdateZCoordinate_Def.hpp (hunk 1 above)
Comment (mperego) [reply src/landIce/evaluators/LandIce_UpdateZCoordinate_Def.hpp:34]: We could allow lossy for building jacobian and sensitivities.
In this case, we need the mesh coords to be FAD type to get an exact Jacobian, but we can also neglect those derivatives and build an approximate Jacobian.

Comment (bartgol) [reply src/landIce/evaluators/LandIce_UpdateZCoordinate_Def.hpp:34]: You mean derivatives of H wrt u and viceversa?

Comment (mperego) [reply src/landIce/evaluators/LandIce_UpdateZCoordinate_Def.hpp:34]: Here we are missing the derivatives of the z-coords of the mesh w.r.t. H.

Comment (mperego) [reply src/landIce/evaluators/LandIce_UpdateZCoordinate_Def.hpp:34]: If you look in that evaluator you'll see a `Albany::convertScalar<MeshScalarT>(zcoord)`. Here `zcoord` is the value of the updated vertical coordinate that depend on `H`, and is a `ScalarT`. If `MeshScalarT` is a double during the `Jacobian` evaluation, we are stripping out the derivatives to be able to assign t to the mesh.

**Review by jewatkins — APPROVED** [review]
lgtm

**Comment by mperego** [comment]
Thanks @bartgol and @jewatkins for reviewing this.

**Pull request merged**